src/rust/bitbox02-rust/src/hww/api/electrum.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::pb;
+use super::Error;
+
+use pb::response::Response;
+
+use bitbox02::keystore;
+
+const HARDENED: u32 = 0x80000000;
+const ELECTRUM_WALLET_ENCRYPTION_KEYPATH_LEVEL_ONE: u32 = 4541509 + HARDENED;
+const ELECTRUM_WALLET_ENCRYPTION_KEYPATH_LEVEL_TWO: u32 = 1112098098 + HARDENED;
```

Comment:
Why is it called "encryption"? It's just an xpub in the way electrum wants it, right. Nothing else special.

Reply:
Electrum uses it as an encryption key to encrypt the wallet.

Reply:
Sure but that's on the Electrum side. Who knows what other uses people will find for xpubs :)
All we do here is formatting an xpub, for what concerns the bb02 API anyway.

Reply:
Ok, maybe the const names are fine but the protobuf request name is strange.

Reply:
>  Sure but that's on the Electrum side. Who knows what other uses people will find for xpubs :)
> All we do here is formatting an xpub, for what concerns the bb02 API anyway.

It was made for only this purpose. I don't see it being helpful to make a name more generic and less useful as a result. Xpub is a misnomer here as you should not make addresses from it and send coins to it (Electrum simply abused xpubs for the purpose).

We don't call xpubs `bytes` either cause it could be used for other purposes :D

Reply:
I see. This one is a very good point:

> you should not make addresses from it and send coins to it (Electrum simply abused xpubs for the purpose)

Would've appreciated if it were somewhere in the docs, maybe for `process` fn.

Reply:
Rebased and added the comment.